File: python/leetcode/452.py
class Solution(object):
    def findMinArrowShots(self, points):
        points = sorted(points, key = lambda x: x[1])
        res, end = 0, -float('inf')
        for point in points:
			if point[0] > end:
				res += 1
				end = point[1]
        return res

File: cpp/leetcode/452.cpp
class Solution {
public:
    int findMinArrowShots(vector<pair<int, int>> &points) {
        sort(points.begin(), points.end(), [](pair<int, int> a, pair<int, int> b) { return a.second < b.second; });
        long long res = 0, end = LONG_MIN;
        for (auto point : points) {
            if (point.first > end) {
                res++;
                end = point.second;
            }
        }
        return res;
    }
};